src/renderer/components/controls/DownloadContentButton.vue

```diff
@@ -7,43 +7,169 @@ SPDX-License-Identifier: MIT
 <template>
     <div class="fullwidth">
         <div class="progress-bar-outer margin-left-md margin-right-md">
-            <MapDownloadProgress :map-name="map?.springName" :height="75"></MapDownloadProgress>
+            <DownloadProgress :map-name="map?.springName" :engine="engineVersion" :game="gameVersion" :height="75"></DownloadProgress>
         </div>
-        <button
-            v-if="map.isInstalled"
-            class="quick-play-button fullwidth"
-            :class="props.class"
-            :disabled="props.disabled"
-            @click="props.onClick"
-        >
+        <button v-if="ready" class="quick-play-button fullwidth" :class="props.class" :disabled="props.disabled" @click="props.onClick">
             <slot />
         </button>
-        <Button v-else-if="map.isDownloading" class="grey quick-download-button fullwidth anchor" @input.stop style="min-height: unset">{{
+        <Button v-else-if="isDownloading" class="grey quick-download-button fullwidth anchor" @input.stop style="min-height: unset">{{
             t("lobby.components.controls.downloadContentButton.downloading")
         }}</Button>
-        <Button v-else class="red fullwidth quick-download-button" @click="downloadMap(map.springName)" style="min-height: unset">{{
-            t("lobby.components.controls.downloadContentButton.download")
-        }}</Button>
+        <Button
+            v-else
+            class="red quick-download-button fullwidth"
+            @click="beginDownload(map?.springName, engineVersion, gameVersion)"
+            style="min-height: unset"
+            >{{ t("lobby.components.controls.downloadContentButton.download") }}</Button
+        >
     </div>
 </template>
 
 <script lang="ts" setup>
+import { computed, ref } from "vue";
 import { MapDownloadData } from "@main/content/maps/map-data";
 import Button from "@renderer/components/controls/Button.vue";
 import { downloadMap } from "@renderer/store/maps.store";
 import { ButtonProps } from "primevue/button";
-import MapDownloadProgress from "@renderer/components/common/MapDownloadProgress.vue";
+import DownloadProgress from "@renderer/components/common/DownloadProgress.vue";
 import { useTypedI18n } from "@renderer/i18n";
+import { downloadEngine } from "@renderer/store/engine.store";
+import { downloadGame } from "@renderer/store/game.store";
+import { computedAsync } from "@vueuse/core";
+import { downloadsStore } from "@renderer/store/downloads.store";
+import { notificationsApi } from "@renderer/api/notifications";
+
 const { t } = useTypedI18n();
 
 export interface Props extends /* @vue-ignore */ ButtonProps {
-    map: MapDownloadData;
+    map?: MapDownloadData;
     disabled?: boolean;
     class?: string;
     onClick?: (event: MouseEvent) => void;
+    engineVersion?: string;
+    gameVersion?: string;
```

Comment:
those probably will need to be all `maps`, `gameVersions` and `engineVersions` to support also the e.g. download for matchmaking.

Reply:
Multiple maps I could see, but will we really need to support multiple engine and game version downloads? Obviously, doing one means we just replicate it for the others, but I can't think of a scenario where it'd be wanted.

Reply:
Multiple game and engine versions allow for staged progressive rollouts and A/B testing.

Reply:
And by the way that's already in the protocol https://github.com/beyond-all-reason/tachyon/blob/bbc1aa4817a72f2fcd0d471fe1ca34148e9fa632/schema/matchmaking/list/response.json#L33-L54